pallets/block-rewards/src/mock.rs

```diff
@@ -0,0 +1,383 @@
+	type ExistentialDeposits = ExistentialDeposits;
+	type MaxLocks = ();
+	type MaxReserves = ();
+	type ReserveIdentifier = [u8; 8];
+	type RuntimeEvent = RuntimeEvent;
+	type WeightInfo = ();
+}
+
+impl pallet_restricted_tokens::Config for Test {
+	type Balance = Balance;
+	type CurrencyId = CurrencyId;
+	type Fungibles = OrmlTokens;
+	type NativeFungible = Balances;
+	type NativeToken = NativeToken;
+	type PreCurrency = cfg_traits::Always;
+	type PreExtrTransfer = cfg_traits::Always;
+	type PreFungibleInspect = pallet_restricted_tokens::FungibleInspectPassthrough;
+	type PreFungibleInspectHold = cfg_traits::Always;
+	type PreFungibleMutate = cfg_traits::Always;
+	type PreFungibleMutateHold = cfg_traits::Always;
+	type PreFungibleTransfer = cfg_traits::Always;
+	type PreFungiblesInspect = pallet_restricted_tokens::FungiblesInspectPassthrough;
+	type PreFungiblesInspectHold = cfg_traits::Always;
+	type PreFungiblesMutate = cfg_traits::Always;
+	type PreFungiblesMutateHold = cfg_traits::Always;
+	type PreFungiblesTransfer = cfg_traits::Always;
+	type PreReservableCurrency = cfg_traits::Always;
+	type RuntimeEvent = RuntimeEvent;
+	type WeightInfo = ();
+}
+
+#[derive(
+	scale_info::TypeInfo, Debug, Copy, codec::Encode, codec::Decode, PartialEq, Clone, MaxEncodedLen,
+)]
+pub enum RewardDomain {
+	Liquidity,
+	Block,
+}
+
+frame_support::parameter_types! {
+	pub const RewardsPalletId: PalletId = PalletId(*b"d/reward");
+	pub const NativeToken: CurrencyId = CurrencyId::Native;
+
+	#[derive(scale_info::TypeInfo)]
+	pub const MaxCurrencyMovements: u32 = 1;
+}
+
+impl pallet_rewards::Config<pallet_rewards::Instance1> for Test {
+	type Currency = Tokens;
+	type CurrencyId = CurrencyId;
+	type DomainId = RewardDomain;
+	type GroupId = u32;
+	type PalletId = RewardsPalletId;
+	type RewardCurrency = NativeToken;
+	type RewardIssuance =
+		pallet_rewards::issuance::MintReward<AccountId, Balance, CurrencyId, Tokens>;
+	type RewardMechanism = pallet_rewards::mechanism::base::Mechanism<
+		Balance,
+		i64,
+		sp_runtime::FixedI128,
+		MaxCurrencyMovements,
+	>;
+	type RuntimeEvent = RuntimeEvent;
+}
+
+// pub type MockRewards =
+// 	cfg_traits::rewards::mock::MockRewards<Balance, u32, (u8, CurrencyId), AccountId>;
+
+frame_support::parameter_types! {
+	#[derive(scale_info::TypeInfo)]
+	pub const MaxGroups: u32 = 1;
+	#[derive(scale_info::TypeInfo, Debug, PartialEq, Clone)]
+	pub const MaxChangesPerSession: u32 = 50;
+	#[derive(scale_info::TypeInfo, Debug, PartialEq, Clone)]
+	pub const MaxCollators: u32 = MAX_COLLATORS;
+	pub const BlockRewardsDomain: RewardDomain = RewardDomain::Block;
+}
+
+impl pallet_block_rewards::Config for Test {
+	type AdminOrigin = EnsureRoot<AccountId>;
+	type AuthorityId = UintAuthorityId;
+	type Balance = Balance;
+	type Beneficiary = RewardRemainderMock;
+	type Currency = Tokens;
+	type Domain = BlockRewardsDomain;
+	type MaxChangesPerSession = MaxChangesPerSession;
+	type MaxCollators = MaxCollators;
+	type Rewards = Rewards;
```

Comment:
@lemunozm I did not go with the mock here since I felt more comfortable staying closer to the runtime configuration. Also felt like the mock in its current form would have bloated up the tests with a lot of boilerplate. 

Now that I can be sure the tests are correct and sufficient, I am happy to revert this if you think it is to be preferred.

Reply:
It's true that the correct mock system (based on `mockall`) is hard and bolierplate. If you anyways needs to use `pallet_restricted_tokens` as a dependency it also loses part of its sense.

My only concern is, what happen if a user uses a different reward mechanism? Are we testing and make assumptions as if it was always the `base` mechanism? Are we not covered if the user of this pallet changes the mechanism?

By now, I think let leave it as it is, and once the `mock-builder` is completed, we'll see if then it simplify things and if it's worth using here 👍🏻